src/groups/mqb/mqbi/mqbi_queue.h

```diff
@@ -1025,6 +1082,52 @@ class QueueHandleFactory {
 //                             INLINE DEFINITIONS
 // ============================================================================
 
+inline const char* InlineResult::toAscii(InlineResult::Enum value)
+{
+#define CASE(X)                                                               \
+    case e_##X: return #X;
+
+    switch (value) {
+        CASE(SUCCESS)
+        CASE(UNAVAILABLE)
+        CASE(CHANNEL_ERROR)
+        CASE(INVALID_PARTITION)
+        CASE(INVALID_PRIMARY)
+        CASE(INVALID_GEN_COUNT)
+        CASE(SELF_PRIMARY)
+    default: return "(* UNKNOWN *)";
+    }
+#undef CASE
+}
+
+inline bool InlineResult::isPermanentError(InlineResult::Enum value)
+{
+    if (value == InlineResult::Enum::e_INVALID_PARTITION ||
+        value == InlineResult::Enum::e_SELF_PRIMARY) {
+        return true;  // RETURN
+    }
+    else {
+        return false;  // RETURN
+    }
+}
+
+inline bmqt::AckResult::Enum
+InlineResult::toAckResult(InlineResult::Enum value)
+{
+    bmqt::AckResult::Enum ackResult = bmqt::AckResult::e_SUCCESS;
+
+    if (value == InlineResult::Enum::e_SUCCESS) {
+        ackResult = bmqt::AckResult::e_SUCCESS;
+    }
+    else if (value == InlineResult::Enum::e_INVALID_PARTITION) {
+        ackResult = bmqt::AckResult::e_INVALID_ARGUMENT;
+    }
+    else {
+        ackResult = bmqt::AckResult::e_UNKNOWN;
+    }
+    return ackResult;
```

Comment:
I think switch is both shorter/simpler, and no intermediate variable
```suggestion
    switch (value) {
        case InlineResult::Enum::e_SUCCESS:
            return bmqt::AckResult::e_SUCCESS;
        case InlineResult::Enum::e_INVALID_PARTITION:
            return bmqt::AckResult::e_INVALID_ARGUMENT;
        default:
            return bmqt::AckResult::e_UNKNOWN;
    }
```